src/codegen/generators/section_names.rs

```diff
@@ -40,6 +40,18 @@ pub fn mangle_type(index: &Index, ty: &typesystem::DataType) -> Result<section_m
                 Ok(acc)
             })?,
         },
+        DataTypeInformation::Array { inner_type_name, .. } => {
+            Type::Array { inner: Box::new(access_inner(inner_type_name)?) }
+        }
+        // FIXME: Is that correct?
+        // For code generation, the actual range does not matter - it is not a breaking change
+        // if a variable's range changes, at least not for codegen, since the underlying type will stay
+        // the same. Therefore, only encode it as its underlying type.
+        DataTypeInformation::SubRange { referenced_type, .. } => access_inner(referenced_type)?,
+        DataTypeInformation::Generic { .. } | DataTypeInformation::Alias { .. } => {
+            // FIXME: Is that correct?
```

Comment:
Yes, generics and aliases will be resolved to the correct types during the resolve phase.